File: pgpainless-sop/src/main/java/org/pgpainless/sop/SOPVImpl.java
// SPDX-FileCopyrightText: 2024 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package org.pgpainless.sop;

import org.jetbrains.annotations.NotNull;
import org.pgpainless.util.ArmoredOutputStreamFactory;
import sop.SOPV;
import sop.operation.DetachedVerify;
import sop.operation.InlineVerify;
import sop.operation.Version;

/**
 * Implementation of the <pre>sopv</pre> interface subset using PGPainless.
 */
public class SOPVImpl implements SOPV {

    static {
        ArmoredOutputStreamFactory.setVersionInfo(null);
    }

    @NotNull
    @Override
    public DetachedVerify detachedVerify() {
        return new DetachedVerifyImpl();
    }

    @NotNull
    @Override
    public InlineVerify inlineVerify() {
        return new InlineVerifyImpl();
    }

    @NotNull
    @Override
    public Version version() {
        return new VersionImpl();
    }
}

File: pgpainless-sop/src/main/kotlin/org/pgpainless/sop/SOPVImpl.kt
// SPDX-FileCopyrightText: 2024 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package org.pgpainless.sop

import org.pgpainless.PGPainless
import org.pgpainless.util.ArmoredOutputStreamFactory
import sop.SOPV
import sop.operation.DetachedVerify
import sop.operation.InlineVerify
import sop.operation.Version

class SOPVImpl(private val api: PGPainless) : SOPV {

    init {
        ArmoredOutputStreamFactory.setVersionInfo(null)
    }

    override fun detachedVerify(): DetachedVerify = DetachedVerifyImpl(api)

    override fun inlineVerify(): InlineVerify = InlineVerifyImpl(api)

    override fun version(): Version = VersionImpl(api)
}
